### PR TITLE
Possibility to clear all records from IndexedDB

### PR DESCRIPTION
**Pull request by polarnis**
While working on a pet project of mine, I found myself in need of clearing the whole store, not only specific records. This PR tries to bring it upstream by using `clear` method of store.
As I'm not very familiar with the codebase, I tried to stay as close as possible to the way IndexedDB API is provided atm (including tests). Hopefully I managed to update everything what's required.

### CODE REVIEW

**Review by kripken — APPROVED** [review]
Code looks good to me! Let's leave this open though for a bit to see if there are more thoughts on the API here.

File: src/library_idbstore.js [hunk 1]
```diff
@@ -125,6 +139,15 @@ var LibraryIDBStore = {
       });
     });
   },
+  emscripten_idb_clear__async: true,
+  emscripten_idb_clear: (db, perror) => {
+    return Asyncify.handleSleep(function(wakeUp) {
+      IDBStore.clearStore(UTF8ToString(db), function(error) {
+        {{{ makeSetValue('perror',  0, '!!error', 'i32') }}};
```
Comment (sbc100) [inline src/library_idbstore.js:146]: Looks like two spaces before the `0`

**Review by sbc100 — APPROVED** [review]

File: src/library_idbstore.js (hunk 1 above)
Comment (sbc100) [reply src/library_idbstore.js:146]: Can you use arrow functions here.  I fixed up the rest of this file in #20200

Comment (polarnis) [reply src/library_idbstore.js]: Sorry for my late response, it should be done now (I hope you don't mind force pushes)

**Review by sbc100 — APPROVED** [review]

**Pull request merged**